### PR TITLE
Allow labeled atoms to have working queries

### PR DESCRIPTION
**Pull request by bp-kelley**
@greglandrum noticed an issue with labeled queries:

```
In [17]: scaff = Chem.MolFromSmarts('[a:1]CC[AR:2]')

In [18]: ms = [Chem.MolFromSmiles(smi) for smi in ('c1ccccc1CCC1CCCCC1','FC1CCCCC1CCc1ccncc1')]

In [19]: rdRGroupDecomposition.RGroupDecompose(scaff,ms,asSmiles=True)
Out[19]: 
([{'Core': 'C(C[*:2])[*:1]',
   'R1': 'c1ccc(CCC(CCC[*:2])[*:1])cc1',
   'R2': 'c1ccc(CCC(CCC[*:2])[*:1])cc1'},
  {'Core': 'C(C[*:2])[*:1]', 'R1': 'c1cc([*:1])ccn1', 'R2': 'FC1CCCCC1[*:2]'}],
 [])
```

This happens because the buildMolMatch code strips out dummy atoms and doesn't properly check for atom queries.

The fix is a bit annoying since we have atom labels which can be isotopes which are converted to queries and need to be removed.  This fix takes care of most of the corner cases but leaves dummy atoms that have non-dummy queries on them.

An easier test case is the following which doesn't properly match the isotopes in the output

```
>>> scaffold = Chem.MolFromSmarts("[$([1C]):1]C[$([2C]):2]")
>>> rgd.RGroupDecompose(scaffold, [Chem.MolFromSmiles(s) for s in ["N[1C]C[2C]", "N[2C]C[1C]"]], asSmiles=True)
([{'Core': 'C([*:1])[*:2]', 'R1': '[2C][*:1]', 'R2': 'N[1C][*:2]'},
  {'Core': 'C([*:1])[*:2]', 'R1': '[1C][*:1]', 'R2': 'N[2C][*:2]'}],
 [])
```

### CODE REVIEW

File: Code/GraphMol/RGroupDecomposition/RGroupCore.cpp [hunk 1]
```diff
@@ -418,6 +418,22 @@ void RCore::buildMatchingMol() {
...
+      if(atom->hasQuery()) {
+        auto q = atom->getQuery()->getDescription();
+	// If we are anything but AtomNull or AtomAtomicNum we are not
+	//  fully a dummy atom
+	if(q != "AtomNull" && q != "AtomAtomicNum") {
+	  unsigned int type=0xFFFFFFFF;
+	  //  If we are were set as an RLABEL of type Isotope we probably
+	  //   have a query of named "AtomIsotope"
+	  //   we DO strip these unless they have an additional query
+	  if(!atom->getPropIfPresent(RLABEL_TYPE, type) || type != RGroupLabels::IsotopeLabels || q != "AtomIsotope") {
```
Comment (greglandrum) [inline Code/GraphMol/RGroupDecomposition/RGroupCore.cpp]: I generally prefer to do the simple checks first in the if:
```suggestion
	  if(q != "AtomIsotope" || !atom->getPropIfPresent(RLABEL_TYPE, type) || type != RGroupLabels::IsotopeLabels) {
```
